### PR TITLE
Feature mes 2461 call get next batch and extract

### PR DESCRIPTION
**Pull request by rossfauldsbjss**
Create a stubbed batch fetcher that fetches results from a url (which will ultimately be the getNextBatch endpoint) uncompresses them and returns an array containing the uncompressed tests.

## Pull Request checklist

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop

### CODE REVIEW

File: src/functions/tarsUpload/framework/adapter/fetch/StubBatchFetcher.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { IBatchFetcher } from '../../../application/secondary/IBatchFetcher';
+import { injectable, inject } from 'inversify';
+import { StandardCarTestCATBSchema } from '@dvsa/mes-test-schema/categories/B';
+import axios, { AxiosInstance } from 'axios';
+import * as zlib from 'zlib';
+import { ITestResultHTTPConfig } from '../upload/ITestResultHTTPConfig';
+import { TYPES } from '../../di/types';
+@injectable()
+export class StubBatchFetcher implements IBatchFetcher {
```
Comment (ghost) [inline src/functions/tarsUpload/framework/adapter/fetch/StubBatchFetcher.ts]: This is called `StubBatchFetcher`, but it's making a real call.

File: src/functions/tarsUpload/framework/adapter/fetch/StubBatchFetcher.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  private endpoint: string;
+  axios: AxiosInstance;
+
+  private nextBatch: StandardCarTestCATBSchema[] = [];
```
Comment (ghost) [inline src/functions/tarsUpload/framework/adapter/fetch/StubBatchFetcher.ts]: `nextBatch` is unused.

File: src/functions/tarsUpload/framework/adapter/fetch/StubBatchFetcher.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  ) {
+    this.axios = axios.create({
+    });
+    this.endpoint = testResultHttpConfig.endpoint;
+  }
+
+  fetchNextUploadBatch(): Promise<StandardCarTestCATBSchema[]> {
+    return new Promise((resolve, reject) => {
+      const result = this.axios.get(this.endpoint);
+      result.then((response) => {
```
Comment (ghost) [inline src/functions/tarsUpload/framework/adapter/fetch/StubBatchFetcher.ts]: If this intended to be a real implementation, it should probably have some error handling for the various things that can go wrong (request errors, not valid b64, not valid gzip, not valid JSON, etc).

**Review by ghost — APPROVED** [review]
Couple of queries on this.

**Review by szabikr — APPROVED** [review]
Once the already raised issues are addressed is good to go :)

**Pull request merged**